### PR TITLE
test: add file/SQL level test for pruning parquet row group with decimal data type.

### PR DESCRIPTION
**Pull request by liukun4515**
# Which issue does this PR close?

This PR just add more test cases for pruning row group with the decimal data type in parquet file.

part of #2962

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by mingmwang** [comment]
Can you please also add test cases to cover the decimal type with large precisions? For example decimal(18 ,0), decimal (38, 0). Maybe the UT will fail but once the getStats methods is fixed for binary array encoded decimal,  the UT should pass.

**Comment by liukun4515** [comment]
> Can you please also add test cases to cover the decimal type with large precisions? For example decimal(18 ,0), decimal (38, 0). Maybe the UT will fail but once the getStats methods is fixed for binary array encoded decimal, the UT should pass.

thanks for your suggestions, and will add test for larger precision of decimal which will be stored as fixed_length_byte_array in the parquet.

**Comment by liukun4515** [comment]
add more test case for other precision.

File: datafusion/core/tests/parquet_pruning.rs [hunk 1]
```diff
@@ -449,6 +450,154 @@ async fn prune_int32_eq_in_list_negated() {
     assert_eq!(output.result_rows, 19, "{}", output.description());
 }
 
+async fn test_prune_decimal(
```
Comment (liukun4515) [inline datafusion/core/tests/parquet_pruning.rs:453]: I'm inspired by this, I will collation other test case and make the test cases cleanly

Comment (alamb) [reply datafusion/core/tests/parquet_pruning.rs:453]: Can you please clarify what you mean by this comment?

I didn't quite understand if you planned to do this combination as part of this PR or if you planned to do it as a follow on PR. 

In other words, is this PR ready for review, or do you plan more work?

**Review by alamb — APPROVED** [review]

File: datafusion/core/tests/parquet_pruning.rs (hunk 1 above)
Comment (liukun4515) [reply datafusion/core/tests/parquet_pruning.rs:453]: Sorry for the confused comments.
When I work on the decimal test, and find all the `pruning test` has the common logic.
I can collation them in the next pr.
```
async test_prune(
    case_type: Scenario,
    sql: &str,
    expected_errors: Option<usize>,
    expected_row_group_pruned: Option<usize>,
    expected_results: usize) -> {}
```
All the test cases such as `prune_f64_lt`,`prune_i64_lt` can use `test_prune` to replace the repeated logic.

**Comment by liukun4515** [comment]
I will merge this pr first, and collation all the code of the `prune test`.

File: datafusion/core/tests/parquet_pruning.rs (hunk 1 above)
Comment (liukun4515) [reply datafusion/core/tests/parquet_pruning.rs:453]: follow up pr: https://github.com/apache/arrow-datafusion/pull/2986

**Pull request merged**